### ISSUE TITLE
Add type hints to function signatures

### ISSUE THREAD

**Issue by volcan01010** [volcan01010/bng#3]
### Summary

As a BNG user, I want type hints on the function signatures so that my IDE can tell me what types to expect.

### Description

Adding type hints (https://realpython.com/lessons/type-hinting/) helps IDEs with autocompletion and gives users information about what inputs they are required to provide.

### Acceptance criteria

+ [ ] `to_osgb36` has type hints on the function signature
+ [ ] `from_osgb36` has type hints on the function signature

**Comment by Jukelyn** [comment]
Did you want just those two functions to have type hints?

**Comment by volcan01010** [comment]
Yes, please. Those are the only ones that end users will interact with.

**Comment by volcan01010** [comment]
Also, don't worry about backwards compatibility too much. The next release
can be for Python >= 3.7.

**Comment by Jukelyn** [comment]
I've done them and they work for newer Python 3 versions! commit above [e714258](https://github.com/Jukelyn/bng/commit/e714258e799c82779c82b12a1a681dff64d9a4e6)

**Comment by volcan01010** [comment]
Thanks!

We need to specify the type for `figs` as well.

Also, can you please update `setup.py` to specify Python 3.7 or greater?

John

**Comment by Jukelyn** [comment]
Done! https://github.com/volcan01010/bng/commit/418f1f03a256bff4d4a8095468d47252bf797b67
